vscode-lean4/src/utils/fsExistHelper.ts

```diff
@@ -0,0 +1,18 @@
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
+import { promises } from 'fs';
+
+
+export async function fsExistHelper(pathFile: string): Promise<boolean> {
+    /*
+    Function that helps to replace fs.existsSync (using existsSync to check for the existence
+    of a file before calling fs.open(), fs.readFile() or fs.writeFile() is not recommended.
+    Doing so introduces a race condition, since other processes may change the file's state between the two calls.
+    Instead, user code should open/read/write the file directly and handle the error raised if the file does not exist.)
+    */
+
+    const fileExists = await promises.access(this._temporaryStorage).then(() => true).catch(() => false);
+    if (!fileExists) {
```

Comment:
Why not `return !fileExists`?

Reply:
Hi Wojciech ,  this PR is a work in progress and I am working with Mariana on it, I'll let you know when it's ready for review, thanks.

Reply:
I see, I started the review now because it has been merged. Please see #136, and I guess the discussion can continue in another PR from the same branch?

Reply:
Yes, the merge was accidental, and I have undone the merge.

Reply:
Yes, let's create a new PR since I don't know how to change the state of this PR back to unmerged.

Reply:
Ah okay, sorry about creating even more confusion. I closed the above, and yeah it is not possible to go back to not-merged so a new PR is needed.